### PR TITLE
Improve coverage script

### PR DESCRIPTION
**Pull request by ionut-arm**
This commit improves the coverage part of ci.sh by excluding a number of
files that we cannot test and removing the PKCS11 provider from the
build.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

### CODE REVIEW

File: ci.sh [hunk 1]
```diff
@@ -174,26 +174,32 @@ if [ "$PROVIDER_NAME" = "trusted-service" ]; then
 fi
 
 if [ "$PROVIDER_NAME" = "coverage" ]; then
-    PROVIDERS="mbed-crypto tpm pkcs11"
-    # Install tarpaulin and start Parsec with it
+    PROVIDERS="mbed-crypto tpm" # pkcs11 not supported because of a segfault when the service stops
```
Comment (hug-dev) [inline ci.sh:177]: Maybe it would be good to open an issue to reference that and link it here!

File: ci.sh [hunk 2]
```diff
@@ -174,26 +174,32 @@ if [ "$PROVIDER_NAME" = "trusted-service" ]; then
...
     for provider in $PROVIDERS; do
+        # Set up run
         PROVIDER_NAME=$provider
         TEST_FEATURES="--features=$provider-provider"
         cp $(pwd)/e2e_tests/provider_cfg/$provider/config.toml $CONFIG_PATH
         mkdir -p reports/$provider
-        RUST_LOG=info cargo tarpaulin --out Xml --forward --command build --output-dir $(pwd)/reports/$provider --features="all-providers,all-authenticators" --run-types bins --timeout 3600 -- -c $CONFIG_PATH &
+
+        # Start service
+        RUST_LOG=info cargo tarpaulin --out Xml --forward --command build --exclude-files="$EXCLUDES" --output-dir $(pwd)/reports/$provider --features="all-providers,all-authenticators" --run-types bins --timeout 3600 -- -c $CONFIG_PATH &
```
Comment (hug-dev) [inline ci.sh:192]: Actually, does it make sense to do `all-providers` then? Should we remove the one we ignore? Are they automatically ignored if they are not compiled in?

Comment (ionut-arm) [reply ci.sh:192]: I just exclude them through the CLI option and they don't get instantiated anyway. Could change it to only compile what we need!

File: ci.sh (hunk 1 above)
Comment (ionut-arm) [reply ci.sh:177]: Yeah, I'll open an issue!

**Pull request merged**